straxen/scada.py

```diff
@@ -0,0 +1,246 @@
+                            end,
+                            parameter_key,
+                            parameter_name,
+                            value_every_seconds=1):
+    """
+    Function to query the values of a single parameter from SCData.
+
+    :param start: Start time in ns unix time
+    :param end: End time in ns unix time
+    :param parameter_key: Key to identify queryed parameter in the
+        DataFrame
+    :param parameter_name: Parameter name in Scada/historian database.
+    :param value_every_seconds: Defines with which time difference
+        values should be returned. Must be an integer!
+        Default: one value per 1 seconds.
+
+    :returns: DataFrame with a time and parameter_key column.
+    """
+    if value_every_seconds < 1:
+        mes = ("Scada takes only values every second. Cannot ask for a"
+               " higher sampling rate than one value per second. However"
+               f" you asked for one value every {value_every_seconds} seconds.")
+        raise ValueError(mes)
+    if not isinstance(value_every_seconds, int):
+        raise ValueError('"value_every_seconds" must be an int!')
+
+    # First we have to create an array where we can fill values with
+    # the sampling frequency of scada:
+    # TODO: Add a check in case user queries to many values. If yes read
+    #  the data in chunks. How much are too many?
+    seconds = np.arange(start, end+1, 10**9)  # +1 to make sure endtime is included
+    df = pd.DataFrame()
+    df.loc[:, 'time'] = seconds
+    df['time'] = df['time'].astype('<M8[ns]')
+    df.set_index('time', inplace=True)
+    
+    # Check if first value is in requested range:
+    query = SCADA_SECRETS.copy()
+    query['name'] = parameter_name
+    query['EndDateUnix'] = (start//10**9) + 1  # +1 since it is end before exclusive
+    query = urllib.parse.urlencode(query)
+    values = requests.get(SCLastValue_URL + query)
+    
+    try:
+        temp_df = pd.read_json(values.text)
+    except:
+        mes = values.text  # returns a dictionary as a string
+        mes = eval(mes)  
+        raise ValueError(f'SCADA raised the following error "{mes["message"]}" '
+                         f'when looking for your parameter "{parameter_name}"')
+    
+    # Store value as first value in our df
+    df.loc[df.index.values[0], parameter_key] = temp_df['value'][0]
+
+    # Query values between start+1 and end time:
+    query = SCADA_SECRETS.copy()
+    query["StartDateUnix"] = (start//10**9) + 1
+    query["EndDateUnix"] = (end//10**9)
+    query['name'] = parameter_name
+    query = urllib.parse.urlencode(query)
+    values = requests.get(SCData_URL + query)
+    
+    try:
+        # Here we cannot do any better since the Error message returned
+        # by the scada api is always the same...
+        temp_df = pd.read_json(values.text)
+        df.loc[temp_df['timestampseconds'], parameter_key] = temp_df.loc[:, 'value'].values
+    except:
```

Comment:
Too broad an exception, can we be a little more specific?

Reply:
Okay done

Reply:
@WenzDaniel I don't think you did not?

Reply:
I do not know what you mean I nailed it down to ValueError?

Reply:
Perhaps I looked at the wrong commit history, there are quite a few ;) but looks good now!